### PR TITLE
feat: add configurable CA certificate in Observability RW Proxy OCP template

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Related to https://issues.redhat.com/browse/MGDSTRM-10333.

This PR updates the Observability Remote Write Proxy OCP Template to allow enabling/disabling the usage of a CA certificate to perform TLS verification during the token verification phase in the Observability Remote Write Proxy. During that phase Observability Remote Write Proxy contacts KFM as its token verifier.

This is useful in those scenarios where the token verifier endpoint is an HTTPs endpoint and it uses a self-signed certificate. For example, when referencing a K8s service as the token verifier endpoint  and self-signed TLS certificates  signed by the OCP cluster are demanded for that K8s service.

This PR also creates a K8s ConfigMap where the CA certificate content is automatically set in it with the `service-ca.crt` key. The PR also updates the template to mount that ConfigMap in the Observability Remote Write Proxy pod so it can access it when CA certificate usage is set for the Observability Remote Write Proxy.

## Verification Steps
1. Deploy Observability Remote Write Proxy template. For how to do it see https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/docs/deploying-observability-remote-write-proxy-in-openshift.md. When doing so, set `OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_SERVICE_SERVING_CA_ENABLED` to true
2. Verify that the Observability Remote Write Proxy pod has the `--token.verification.cacert.enabled` is set to true.
3. Verify that a ConfigMap named `observability-remote-write-proxy-service-serving-ca` has been created and it contains a `service-ca.crt` key with content
4. Verify that the Observability Remote Write Proxy pod has a file in it named `/obs-rw-proxy-service-serving-ca/service-ca.crt` with the same content as the value of the ConfigMap mentione above

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: templates/observability-remote-write-proxy.yml [hunk 1]
```diff
@@ -260,6 +291,44 @@ parameters:
...
+    specified in the file pointed by
+    OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_SERVICE_SERVING_CA_FILEDIR/OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_SERVICE_SERVING_CA_FILENAME
+    is appended to the system's certificate trusted store when token verification
+    is performed. See OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_ENABLED
+    for details on token verification.
+    The Serivce Serving CA certificate has to be a X.509 certificate encoded
+    in PEM format.
+    Only takes effect when OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_ENABLED
+    is set to true.
+  value: "false"
```
Comment (machi1990) [inline templates/observability-remote-write-proxy.yml:306]: should we default to `true`?

File: Makefile [hunk 2]
```diff
@@ -994,6 +994,7 @@ deploy/observability-remote-write-proxy: OBSERVABILITY_REMOTE_WRITE_PROXY_FORWAR
 deploy/observability-remote-write-proxy: OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_ENABLED ?= "true"
 deploy/observability-remote-write-proxy: OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_URL ?= ""
 deploy/observability-remote-write-proxy: OBSERVABILITY_REMOTE_WRITE_PROXY_OIDC_ENABLED ?= "true"
+deploy/observability-remote-write-proxy: OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_SERVICE_SERVING_CA_ENABLED ?= "false"
```
Comment (machi1990) [inline Makefile:997]: same comment as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1610/files#r1124463526

File: templates/observability-remote-write-proxy.yml (hunk 1 above)
Comment (miguelsorianod) [reply templates/observability-remote-write-proxy.yml:306]: No. `false` is enough

File: Makefile (hunk 2 above)
Comment (miguelsorianod) [reply Makefile:997]: Replied there

File: templates/observability-remote-write-proxy.yml (hunk 1 above)
Comment (machi1990) [reply templates/observability-remote-write-proxy.yml:306]: Make sense.

In that case, I suggest we default to the KFM route https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/0d7ca274d5133af95a897140133c133b83e43dbe/Makefile#L995 to the deployed KFM route?

something like https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/0d7ca274d5133af95a897140133c133b83e43dbe/Makefile#L898 + the token verification endpoint

What do you think?

**Review by machi1990 — APPROVED** [review]
lgtm

I left a suggestion regarding some defaults.

File: templates/observability-remote-write-proxy.yml (hunk 1 above)
Comment (miguelsorianod) [reply templates/observability-remote-write-proxy.yml:306]: This assumes that KFM has been deployed beforehand.
I prefer to keep that independent.

**Pull request merged**